_posts/2023-03-06-NGINX-Dashboards-with-Fluent-Bit.md

```diff
@@ -0,0 +1,68 @@
+
+[Lua filters](https://docs.fluentbit.io/manual/pipeline/filters/lua) give users extreme flexibility in how they transform their data, including modification, addition, enrichment via API, calculation, or even redaction. To connect Fluent Bit and the new Simple Schema for Observability, we opted for a Lua script that you can paste into your configuration file to run on top of NGINX-ingested logs.
+
+The following is an example fluent-bit.conf for VMs (VM is an industry-standard abbreviation for virtual machine) or standalone deployments:
+
+```bash
+[INPUT]
+    name tail
+    read_from_head true
+    exit_on_eof true
+    path /tmp/data.log
+    parser nginx
+
+[Filter]
+    Name    lua
+    Match   *
+    code    function cb_filter(a,b,c)local d={}local e=os.date("!%Y-%m-%dT%H:%M:%S.000Z")d["observerTime"]=e;d["body"]=c.remote.." "..c.host.." "..c.user.." ["..os.date("%d/%b/%Y:%H:%M:%S %z").."] \""..c.method.." "..c.path.." HTTP/1.1\" "..c.code.." "..c.size.." \""..c.referer.."\" \""..c.agent.."\""d["trace_id"]="102981ABCD2901"d["span_id"]="abcdef1010"d["attributes"]={}d["attributes"]["data_stream"]={}d["attributes"]["data_stream"]["dataset"]="nginx.access"d["attributes"]["data_stream"]["namespace"]="production"d["attributes"]["data_stream"]["type"]="logs"d["event"]={}d["event"]["category"]={"web"}d["event"]["name"]="access"d["event"]["domain"]="nginx.access"d["event"]["kind"]="event"d["event"]["result"]="success"d["event"]["type"]={"access"}d["http"]={}d["http"]["request"]={}d["http"]["request"]["method"]=c.method;d["http"]["response"]={}d["http"]["response"]["bytes"]=tonumber(c.size)d["http"]["response"]["status_code"]=c.code;d["http"]["flavor"]="1.1"d["http"]["url"]=c.path;d["communication"]={}d["communication"]["source"]={}d["communication"]["source"]["address"]="127.0.0.1"d["communication"]["source"]["ip"]=c.remote;return 1,b,d end
+    call    cb_filter
+
+[OUTPUT]
+    format json
+    name stdout
+    match *
+```
+
+## Importing the dashboard in OpenSearch
+
+See the following tutorials:
+
+-	To see example pre-loaded data, see the following demo. Use the text in the following [readme file](https://github.com/opensearch-project/observability/blob/e18cf354fd7720a6d5df6a6de5d53e51a9d43127/integrations/nginx/samples/preloaded/README.md).
+-	To see an example live NGINX->fluent-bit->OpenSearch workflow, see the following demo. Use the text in the following [readme file](https://github.com/opensearch-project/observability/blob/9267012051fabfc2a971493bddde60448bc48ecf/integrations/nginx/test/README.md).
+- See the following [playground](https://observability.playground.opensearch.org/app/dashboards#/view/96847220-5261-44d0-89b4-65f3a659f13a) that displays the pre-loaded NGINX demo based on fluent translated data ingested into the simple schema logs data-stream.
```

Comment:
```suggestion
- See the following [playground](https://observability.playground.opensearch.org/app/dashboards#/view/96847220-5261-44d0-89b4-65f3a659f13a) that displays the preloaded NGINX demo based on fluent translated data ingested into the simple schema logs data stream.
```

Reply:
Do we mean OpenSearch Playground (capitalized)? This sentence needs some revision toward the end. I'm not following "based on fluent translated data ingested into the simple schema logs data-stream." There's got to be some missing punctuation and capitalization here.